### ISSUE TITLE
Error while someone took my offer

### ISSUE THREAD

**Issue by dwknch** [bisq-network/bisq#2924]
I missed to take screenshots of the error message, but it was along the lines that the price was off the allowed range. However I can provide the log file.
[bisq.log](https://github.com/bisq-network/bisq/files/3334337/bisq.log)
Later I got another warning and error regarding locked funds, which I'm going to report separately

**Comment by a123b** [comment]
Looking at the log file, this seems to be a duplicate of #2910.

**Comment by sqrrm** [comment]
This one should be fixed by https://github.com/bisq-network/bisq/pull/2917

**Comment by a123b** [comment]
I don't think so, this is the error from the log file:
```Juni-27 11:46:46.550 [JavaFX Application Thread] INFO  b.c.t.p.TradeProtocol: Send AckMessage for PayDepositRequest to peer null. tradeId=GOMFNLRZ-48fd1240-ce6f-43b6-98f4-bb622bbbfe95-112, sourceUid=3c77447d-97b3-4353-ad2b-ca408d587432 
Juni-27 11:46:46.565 [JavaFX Application Thread] ERROR b.c.t.TaskRunner: Task failed: MakerProcessPayDepositRequest / errorMessage: Error at taskRunner: PeerAddress must not be null (sendEncryptedMailboxMessage) 
Juni-27 11:46:46.565 [JavaFX Application Thread] ERROR b.c.t.p.TradeProtocol: Error at taskRunner: PeerAddress must not be null (sendEncryptedMailboxMessage) 
Juni-27 11:46:46.565 [JavaFX Application Thread] INFO  b.c.t.p.TradeProtocol: Send AckMessage for PayDepositRequest to peer null. tradeId=GOMFNLRZ-48fd1240-ce6f-43b6-98f4-bb622bbbfe95-112, sourceUid=3c77447d-97b3-4353-ad2b-ca408d587432 
Juni-27 11:46:46.567 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: Uncaught Exception from thread JavaFX Application Thread 
Juni-27 11:46:46.567 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: throwableMessage= PeerAddress must not be null (sendEncryptedMailboxMessage) 
Juni-27 11:46:46.567 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: throwableClass= class java.lang.NullPointerException 
Juni-27 11:46:46.574 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: Stack trace:
java.lang.NullPointerException: PeerAddress must not be null (sendEncryptedMailboxMessage)
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:787)
	at bisq.network.p2p.P2PService.sendEncryptedMailboxMessage(P2PService.java:548)
	at bisq.core.trade.protocol.TradeProtocol.sendAckMessage(TradeProtocol.java:205)
	at bisq.core.trade.protocol.TradeProtocol.handleTaskRunnerFault(TradeProtocol.java:171)
	at bisq.core.trade.protocol.SellerAsMakerProtocol.lambda$handleTakeOfferRequest$2(SellerAsMakerProtocol.java:123)
	at bisq.common.taskrunner.TaskRunner.handleErrorMessage(TaskRunner.java:92)
	at bisq.common.taskrunner.TaskRunner.next(TaskRunner.java:72)
	at bisq.common.taskrunner.TaskRunner.run(TaskRunner.java:60)
	at bisq.core.trade.protocol.SellerAsMakerProtocol.handleTakeOfferRequest(SellerAsMakerProtocol.java:141)
	at bisq.core.trade.SellerAsMakerTrade.handleTakeOfferRequest(SellerAsMakerTrade.java:105)
	at bisq.core.trade.TradeManager.handlePayDepositRequest(TradeManager.java:346)
	at bisq.core.trade.TradeManager.lambda$new$0(TradeManager.java:180)
	at bisq.network.p2p.P2PService.lambda$onMessage$8(P2PService.java:410)
	at java.base/java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:804)
	at java.base/java.util.concurrent.CopyOnWriteArraySet.forEach(CopyOnWriteArraySet.java:425)
	at bisq.network.p2p.P2PService.onMessage(P2PService.java:409)
```
It looks different from the one that's fixed by #2917.

@sqrrm Can you take a look at my comment in #2910 regarding this error?

**Comment by devinbileck** [comment]
I agree with @a123b. I fixed #2917 while investigating #2910, but didn't finish finding the cause of the null peer address. Thanks for your detailed investigation!

**Comment by sqrrm** [comment]
Indeed, I got that wrong.
